paasta_tools/mesos_tools.py

```diff
@@ -655,21 +667,34 @@ def terminate_framework(framework_id):
     resp.raise_for_status()
 
 
-def get_container_name(app_id, slave_hostname=None, task_id=None):
+def get_tasks_from_app_id(app_id, slave_hostname=None):
     tasks = get_running_tasks_from_active_frameworks(app_id)
-    try:
-        if task_id:
-            task = next(task for task in tasks if task['id'] == task_id)
-        elif slave_hostname:
-            task = next(task for task in tasks if slave_hostname in task.slave['hostname'])
-        else:
-            raise TaskNotFound("Couldn't find task")
-    except StopIteration:
-        raise TaskNotFound("Couldn't find task")
-    container_name = "mesos-{0}.{1}".format(task.executor['tasks'][0]['slave_id'],
-                                            task.executor['container'])
-    return container_name
+    if slave_hostname:
+        tasks = filter_tasks(tasks, filter_task_by_hostname, hostname=slave_hostname)
+    return tasks
+
+
+def get_task(task_id, app_id=''):
+    tasks = get_running_tasks_from_active_frameworks(app_id)
+    tasks = filter_tasks(tasks, filter_task_by_task_id, task_id=task_id)
+    if len(tasks) < 1:
+        raise TaskNotFound("Couldn't find task for given id: {0}".format(task_id))
+    if len(tasks) > 1:
+        raise TooManyTasks("Found more than one task with id: {0}, this should not happen!".format(task_id))
+    return tasks[0]
+
+
+def filter_task_by_task_id(task, task_id):
+    return task['id'] == task_id
+
+
+def filter_task_by_hostname(task, hostname):
+    return hostname in task.slave['hostname']
```

Comment:
I think what I actually mean is `.startswith` (trying to support users typing short hostnames).